src/vos/vos_aggregate.c

```diff
@@ -1603,6 +1603,40 @@ delete_evt_entry(struct vos_obj_iter *oiter, vos_iter_entry_t *entry,
 	return rc;
 }
 
+static bool
+handle_deleted(daos_handle_t ih, struct agg_merge_window *mw,
+	       vos_iter_entry_t *entry, unsigned int *acts, int *rc)
+{
+	struct vos_obj_iter	*oiter = vos_hdl2oiter(ih);
+	struct evt_extent	 ext;
+
+	*rc = 0;
+
+	if ((entry->ie_vis_flags & VOS_VIS_FLAG_DELETED) == 0)
+		return false;
+
+	recx2ext(&entry->ie_recx, &ext);
+
+	/** If the new record is a eleged record, go ahead and flush
+	 *  the window first because that's all we have left.
+	 */
+	*rc = flush_merge_window(ih, mw, acts);
```

Comment:
It's more clear to call this flush_merge_window() in following "if (entry->ie_vis_flags & VOS_VIS_FLAG_LAST)"?